lib/matplotlib/transforms.py

```diff
@@ -1670,7 +1610,7 @@ def __array__(self, *args, **kwargs):
         return self.get_matrix()
 
     def __eq__(self, other):
-        if getattr(other, "is_affine", False):
+        if getattr(other, "is_affine", False) and hasattr(other, "get_matrix"):
```

Comment:
This might be right, but I'm not sure what it has to do with this PR...

Reply:
This used to be needed because https://github.com/matplotlib/matplotlib/pull/14725/files#diff-9d3a2ed9922d92407a1b5e9c4862d1d1L221 would sometimes compare an Affine2D with a non-Affine2D transform (triggering an AttributeError).  Admittedly this has become unnecessary for the strict purposes of this patch because I now just compare by id() (which is anyways better in case there are two subtransforms which happens to be equal...) but I didn't bother reverting it after the rewrite of write_graphviz.  I can, if you prefer.

Reply:
I'd rather we keep this strictly about moving the debug code, and since it doesn't need this change, better to do it separately, if at all.

Reply:
ok